src/components/structures/RoomView.tsx

```diff
@@ -1319,6 +1319,13 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
     }
 
     private onSearch = (term: string, scope: SearchScope) => {
+        // if we search for a user mxId or an organisation mxId, we need to
+        // search with only the userId or the organizationId.
+        const regex = /^@.*:|^#.*:/;
```

Comment:
We should be trying to convert user IDs within the query to just their localparts, to fix issues where someone searches for `delete request @user:example.org` (for example) - the user ID will be in the middle of the query and still cause breakage. 

I believe we already have a regex somewhere which does this searching, it just needs to be found.